test/modules/TestResultCoordinator/report/CountingReportGenerator.cs

```diff
@@ -71,137 +59,94 @@ sealed class CountingReportGenerator : ITestResultReportGenerator
         /// <returns>Test Result Report.</returns>
         public async Task<ITestResultReport> CreateReportAsync()
         {
-            Logger.LogInformation($"Start to generate report by {nameof(CountingReportGenerator)} for Sources [{this.expectedSource}] and [{this.actualSource}]");
```

Comment:
The idea to use squentialStore in this class is because I don't want to read all the results into memory; use squentialStore to read by batch and verify the result.  If it doesn't apply to Twin scenario, please create a new report generator for Twin.

Reply:
The SimpleResults implementation is using sequentialStore so it doesn't load all results into memory. Same is valid for twin, but not for all cases, for example when it verifies the reported properties it does GetTwin once at the end. I think the counting report it still applies, only the way it retrieves the test results is different.